plugins/inputs/interrupts/interrupts.go

```diff
@@ -90,14 +93,72 @@ scan:
 		} else if len(fields) > cpucount {
 			irq.Type = strings.Join(fields[cpucount+1:], " ")
 		}
+
+		// collect spurious interrupt data for this irq.ID
+		irq.SpuriousCount, irq.SpuriousUnhandled = parseSpurious(filepath.Join("/proc/irq", irq.ID, "spurious"))
 		irqs = append(irqs, *irq)
 	}
 	if scanner.Err() != nil {
 		return nil, fmt.Errorf("Error scanning file: %s", scanner.Err())
 	}
+
+	// For some Linux systems there can be fixed, large number of CPUs reported in
+	// the `/proc/softirqs` file. This number could be much larger than the actual number of
+	// CPUs in the system. The fields for these phantom CPUs contain zeroes. The approach
+	// taken to remove these phantom CPUs is to remove the columns containing all zeros
+	// to the right (higher CPU numbers). For systems where CPUs are dynamically enabled,
+	// this can lead to CPUs not being reported until enabled. However, this is preferable
+	// to collecting metrics for tens or hundreds of phantom CPUs. This cleanup is done in
+	// two steps:
```

Comment:
Should we maybe add a flag that enables/disables this logic? It also changes the past behavior, so we would need to guard it by a flag anyway.

Reply:
I'm fine with adding a flag. But don't expect anyone to have relied upon its former behaviour because the columns are just filled with zero. For example, on my VM with 2 "CPUs" the number of CPUs reported by /proc/softirqs is 128 -- or 126 columns of zeros. There is nothing useful about that information.

So if we need a flag, I'd like to default it to the new behaviour.

Reply:
Can there be a case where those zeros are _actual_ values?